### PR TITLE
Controller: Add other mdns filters

### LINKED ISSUE #6691: Add remaining commissionable node subtype filters
#### Problem
Not all commissinable node subtype filters are implemented or plumbed through to the controller layer.

 #### Proposed Solution
Add filters and plumbing.

### PR DESCRIPTION
**Pull request by cecille**
#### Problem
Mdns exposes serveral subtypes that can be used to narrow the dnssd
search parameters. Previously, the only exposed parameter was the
long discriminator, because it was used in the QR code. 
Fixes #6691

#### Change overview
Adding remaining commisisonable nodes discovery filters. 
Also changes the controller API function to take the filter structure
directly, and uses the python script binding to set the filters
as required.

#### Testing
Tested with M5 in commissionable mode / chip-device-ctrl
- discover usage (discover with no args)
- discover -all
- discover -qr with good and bad qr
- all variants with good and bad values
verified all args with correct values return data, all args with bad arguments return no results.

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/controller/python/chip-device-ctrl.py [hunk 1]
```diff
@@ -516,10 +516,33 @@ def do_resolve(self, line):
...
+            count = count + 0.2
+
+    def wait_for_many_discovered_devices(self):
+        # Discovery happens through mdns, which means we need to wait for responses to come back.
+        # TODO(cecille): I suppose we could make this a command line arg. Or Add a callback when
+        # x number of responses are received. For now, just 2 seconds. We can all wait that long.
+        print("Waiting for device responses...")
+        time.sleep(2)
+
     def do_discover(self, line):
```
Comment (msandstedt) [inline src/controller/python/chip-device-ctrl.py:536]: Although not merged, https://github.com/project-chip/connectedhomeip/pull/7432 has introduced use of argparse to remove the need for all this boilerplate just to define the args interface.

This PR is expanding the number of arguments for do_discover from 2 to 8, so is nearly a complete refactor.  Given this, would you consider using argparse as well here?  It's just a whole lot easier.

**Review by andy31415 — APPROVED** [review]

File: src/controller/python/chip-device-ctrl.py (hunk 1 above)
Comment (cecille) [reply src/controller/python/chip-device-ctrl.py:537]: Changed over in next commit. I'm not going to tackle any other refactors in this PR. I suppose this is ok - it's a bit longer, but does essentially the same thing.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/controller/python/chip-device-ctrl.py (hunk 1 above)
Comment (msandstedt) [reply src/controller/python/chip-device-ctrl.py:537]: Thanks!

**Review by msandstedt — APPROVED** [review]

**Pull request merged**